integration-tests/src/test/java/tachyon/client/keyvalue/KeyValueStoresIntegrationTest.java

```diff
@@ -261,4 +268,146 @@ public void putKeyValueTooLargeTest() throws Exception {
+   * {@link Constants#MB}.
+   *
+   * @param partitionNumber the number of partitions
+   * @param keyValuePairs the key-value pairs in the store, null if you don't want to know them
+   * @return the URI to the created store
+   */
+  private TachyonURI createStoreOfMultiplePartitions(int partitionNumber,
+      List<KeyValuePair> keyValuePairs) throws Exception {
+    // These sizes are carefully selected, one partition holds only one key-value pair.
+    final long maxPartitionSize = Constants.MB; // Each partition is at most 1 MB
+    ClientContext.getConf().set(Constants.KEY_VALUE_PARTITION_SIZE_BYTES_MAX,
+        String.valueOf(maxPartitionSize));
+    final int keyLength = 4; // 4Byte key
+    final int valueLength = 500 * Constants.KB; // 500KB value
+
+    TachyonURI storeUri = new TachyonURI(PathUtils.uniqPath());
+    mWriter = sKVStores.create(storeUri);
+    for (int i = 0; i < partitionNumber; i ++) {
+      byte[] key = BufferUtils.getIncreasingByteArray(i, keyLength);
+      byte[] value = BufferUtils.getIncreasingByteArray(i, valueLength);
+      mWriter.put(key, value);
+      if (keyValuePairs != null) {
+        keyValuePairs.add(new KeyValuePair(key, value));
+      }
+    }
+    mWriter.close();
+
+    Assert.assertEquals(partitionNumber, getPartitionNumber(storeUri));
+
+    return storeUri;
+  }
+
+  /**
+   * Tests that a store of vairous sizes(including empty store) can be correctly deleted.
+   */
+  @Test
+  public void deleteStoreTest() throws Exception {
+    List<TachyonURI> storeUris = Lists.newArrayList();
+    storeUris.add(createStoreOfSize(0, null));
+    storeUris.add(createStoreOfSize(2, null));
+    storeUris.add(createStoreOfMultiplePartitions(3, null));
+
+    for (TachyonURI storeUri : storeUris) {
+      sKVStores.delete(storeUri);
+
+      mThrown.expect(TachyonException.class);
+      sKVStores.open(storeUri);
+    }
+  }
+
+  /**
+   * Tests that two stores of vairous sizes(including empty store) can be correctly merged.
+   */
+  @Test
+  public void mergeStoreTest() throws Exception {
+    List<TachyonURI> storeUris = Lists.newArrayList();
+    List<List<KeyValuePair>> keyValuePairs = Lists.newArrayList();
+
+    List<KeyValuePair> pairs = Lists.newArrayList();
+    storeUris.add(createStoreOfSize(0, pairs));
+    keyValuePairs.add(pairs);
+
+    pairs = Lists.newArrayList();
+    storeUris.add(createStoreOfSize(2, pairs));
+    keyValuePairs.add(pairs);
+
+    pairs = Lists.newArrayList();
+    storeUris.add(createStoreOfMultiplePartitions(3,pairs));
+    keyValuePairs.add(pairs);
+
+    int numStoreUri = storeUris.size();
+    for (int i = 0; i < numStoreUri; i ++) {
+      for (int j = i + 1; j < numStoreUri; j ++) {
+        TachyonURI store1 = storeUris.get(i);
+        TachyonURI store2 = storeUris.get(j);
+
+        sKVStores.merge(store1, store2);
+
+        // store1 no longer exists, because it has been merged into store2.
+        mThrown.expect(TachyonException.class);
```

Comment:
again, `mThrown.expect` will end this test once it sees the expected exception, so you may want to do `try-catch`